### PR TITLE
Add implementation of CsrfViewInjectionInterface.

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Comment by xepozz** [comment]
I'm completely  sure this package shouldn't depend on yii-web

**Comment by vjik** [comment]
Yii-web need only for Csrf::REQUEST_NAME. Replace this to string it's ok?

**Comment by samdark** [comment]
@vjik yes.

**Comment by samdark** [comment]
@xepozz why?

**Comment by xepozz** [comment]
@samdark this package don't need the main functionality of yii-web

**Review by xepozz — APPROVED** [review]

File: src/Csrf/CsrfViewInjection.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    public function __construct(UrlMatcherInterface $urlMatcher)
+    {
+        $this->urlMatcher = $urlMatcher;
+    }
+
+    public function withRequestAttribute(?string $requestAttribute = null): CsrfViewInjectionInterface
+    {
+        $clone = clone $this;
+        $clone->requestAttribute = $requestAttribute ?? self::DEFAULT_REQUEST_ATTRIBUTE;
```
Comment (rustamwin) [inline src/Csrf/CsrfViewInjection.php:36]:
```suggestion
    public function withRequestAttribute(string $requestAttribute): CsrfViewInjectionInterface
    {
        $clone = clone $this;
        $clone->requestAttribute = $requestAttribute;
```

Comment (vjik) [reply src/Csrf/CsrfViewInjection.php:36]: This is not convenient to use in this case: https://github.com/yiisoft/yii-demo/blob/master/src/Contact/ContactController.php#L62

Comment (rustamwin) [reply src/Csrf/CsrfViewInjection.php:36]: maybe it's better?
```suggestion
    public function withRequestAttribute(string $requestAttribute = self::DEFAULT_REQUEST_ATTRIBUTE): CsrfViewInjectionInterface
    {
        $clone = clone $this;
        $clone->requestAttribute = $requestAttribute;
```

Comment (vjik) [reply src/Csrf/CsrfViewInjection.php:36]: This is method from interface `CsrfViewInjectionInterface`, where not `self::DEFAULT_REQUEST_ATTRIBUTE`.

File: src/Csrf/CsrfViewInjectionInterface.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 declare(strict_types=1);
 
-namespace Yiisoft\Yii\View;
+namespace Yiisoft\Yii\View\Csrf;
 
 interface CsrfViewInjectionInterface
```
Comment (samdark) [inline src/Csrf/CsrfViewInjectionInterface.php:7]: Do we need this interface? There's a single implementation and no use of the interface.

File: src/Csrf/CsrfViewInjection.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Yiisoft\Yii\View\Csrf;
+
+use Yiisoft\Router\UrlMatcherInterface;
+use Yiisoft\Yii\View\ContentParamsInjectionInterface;
+use Yiisoft\Yii\View\LayoutParamsInjectionInterface;
+use Yiisoft\Yii\View\MetaTagsInjectionInterface;
+
+class CsrfViewInjection implements
```
Comment (samdark) [inline src/Csrf/CsrfViewInjection.php:10]:
```suggestion
final class CsrfViewInjection implements
```

File: src/Csrf/CsrfViewInjection.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public function withMetaAttribute(string $metaAttribute): self
+    {
+        $clone = clone $this;
+        $clone->metaAttribute = $metaAttribute;
+        return $clone;
+    }
+
+    public function getContentParams(): array
```
Comment (samdark) [inline src/Csrf/CsrfViewInjection.php:54]: How about renaming `params` to `parameters`?

File: src/Csrf/CsrfViewInjection.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public function withRequestAttribute(?string $requestAttribute = null): CsrfViewInjectionInterface
+    {
+        $clone = clone $this;
+        $clone->requestAttribute = $requestAttribute ?? self::DEFAULT_REQUEST_ATTRIBUTE;
+        return $clone;
+    }
+
+    public function withParameter(string $parameter): self
```
Comment (samdark) [inline src/Csrf/CsrfViewInjection.php:40]: It's not clear from the name that this is to set parameter name and not the value.

File: src/Csrf/CsrfViewInjection.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        $clone = clone $this;
+        $clone->parameter = $parameter;
+        return $clone;
+    }
+
+    public function withMetaAttribute(string $metaAttribute): self
```
Comment (samdark) [inline src/Csrf/CsrfViewInjection.php:47]: It's not clear from the name that this is to set attribute name and not the value.

File: src/Csrf/CsrfViewInjection.php [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    private string $parameter = self::DEFAULT_PARAMETER;
+
+    private ?string $csrfToken = null;
+
+    public function __construct(UrlMatcherInterface $urlMatcher)
+    {
+        $this->urlMatcher = $urlMatcher;
+    }
+
+    public function withRequestAttribute(?string $requestAttribute = null): CsrfViewInjectionInterface
```
Comment (samdark) [inline src/Csrf/CsrfViewInjection.php:33]: It's not clear from the name that this is to set attribute name and not the value.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Csrf/CsrfViewInjection.php (hunk 7 above)
Comment (vjik) [reply src/Csrf/CsrfViewInjection.php:33]: Instead `requestAttribute` use `requestAttributeName` in function and variable names?

File: src/Csrf/CsrfViewInjection.php (hunk 5 above)
Comment (vjik) [reply src/Csrf/CsrfViewInjection.php:40]: Instead `parameter` use `parameterName` in function and variable names?

File: src/Csrf/CsrfViewInjection.php (hunk 6 above)
Comment (vjik) [reply src/Csrf/CsrfViewInjection.php:47]: Instead `metaAttribute` use `metaAttributeName` in function and variable names?

**Comment by samdark** [comment]
@vjik should this one be adjusted since https://github.com/yiisoft/csrf is now merged?

**Comment by vjik** [comment]
> @vjik should this one be adjusted since https://github.com/yiisoft/csrf is now merged?

See https://github.com/yiisoft/csrf/issues/2

**Comment by vjik** [comment]
Done. Added usage yiisoft/csrf for get CSRF token.

**Comment by vjik** [comment]
Success tested on yii-demo.

**Review by samdark — APPROVED** [review]

**Pull request merged**